src/module/combat/websocket/ws-combat/util/getSelectedToken.ts

```diff
@@ -0,0 +1,20 @@
+import { ABFDialogs } from "../../../../dialogs/ABFDialogs";
+
+export function getSelectedToken(game: Game): TokenDocument {
+  const selectedTokens = game.canvas.tokens?.controlled || [];
+
+  switch (selectedTokens.length) {
+    case 1:
+      return selectedTokens[0].document;
+
+    case 0:
+      let msg = game.i18n.localize('macros.combat.dialog.error.noSelectedToken.title');
+      ABFDialogs.prompt(msg);
+      throw new Error(msg);
+
+    default:
+      msg = game.i18n.localize('macros.combat.dialog.error.multipleSelectedToken.title');
+      ABFDialogs.prompt(msg);
+      throw new Error(msg);
+  }
```

Comment:
Totalmente opcional:

```suggestion
const selectedTokens = game.canvas.tokens?.controlled ?? [];

if (selectedTokens.length !== 1) {
    const msg = game.i18n.localize(selectedTokens.length > 0 ? 'macros.combat.dialog.error.multipleSelectedToken.title' : 'macros.combat.dialog.error.noSelectedToken.title');

    ABFDialogs.prompt(msg);
    throw new Error(msg);
}

return selectedTokens[0].document;
```

Reply:
@Eryx5502 

Las sugerencias normalmente las suelo poner a modo "demostrativo", es mejor no commitear directamente ya que github no respeta las reglas del repositorio en cuanto indentación, además, que estaríamos commiteando código que no está probado 😢

Reply:
Ay ay ay, tienes obviamente toda la razón. Este soy yo haciendo el noob, algo que tenía que pasar antes o después. Lo siento😔